Templating/LazyFilterRuntime.php

```diff
@@ -57,33 +63,49 @@ public function filterCache(string $path, string $filter, array $config = [], st
         if (\count($config)) {
             $path = $this->cache->getRuntimePath($path, $config);
         }
-        $path = $this->cache->resolve($path, $filter, $resolver);
+        $resolvedPath = $this->cache->resolve($path, $filter, $resolver);
 
-        return $this->appendAssetVersion($path);
+        return $this->appendAssetVersion($resolvedPath, $path);
     }
 
     private function cleanPath(string $path): string
     {
-        if (!$this->assetVersion) {
+        if (!$this->assetVersion && !$this->jsonManifest) {
             return $path;
         }
 
-        $start = mb_strrpos($path, $this->assetVersion);
-        if (mb_strlen($path) - mb_strlen($this->assetVersion) === $start) {
-            return rtrim(mb_substr($path, 0, $start), '?');
+        if ($this->assetVersion) {
+            $start = mb_strrpos($path, $this->assetVersion);
+            if (mb_strlen($path) - mb_strlen($this->assetVersion) === $start) {
+                return rtrim(mb_substr($path, 0, $start), '?');
+            }
+        } elseif ($this->jsonManifest) {
```

Comment:
could we simplify this? we know json manifest is set at this point, it could be `else`.
or it could be simply `if`, if we want to still do this if assetVersion was set but the strlen / start check abvoe did not result in an early return.

Reply:
Specifying both static and json manifest will give a configuration error:
`Invalid configuration for path "framework.assets": You cannot use both "version" and "json_manifest_path" at the same time under "assets".`

It is also possible to group your assets into packages, and then have each of the packages use a different strategy. This however resolves correctly.

Reply:
cool!